src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -174,6 +183,78 @@ public void Serialize(object value, ObjectWriter writer, CancellationToken cance
         }
     }
 
+    private static void Write(ObjectWriter writer, ImmutableDictionary<string, StructuredAnalyzerConfigOptions> optionsByLanguage)
+    {
+        // Only serialize options for C#/VB since other languages are not OOP.
+
+        var csOptions = ImmutableDictionary.GetValueOrDefault(optionsByLanguage, LanguageNames.CSharp);
+        var vbOptions = ImmutableDictionary.GetValueOrDefault(optionsByLanguage, LanguageNames.VisualBasic);
+
+        writer.WriteCompressedUInt((uint)((csOptions != null ? 1 : 0) + (vbOptions != null ? 1 : 0)));
```

Comment:
we have just writeint32 bte, which would make this slightly nicer :)  we really don't benefit much from saving a couple of bytes here.